core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -99,6 +99,15 @@ object Consumer {
     Source.fromGraph(ConsumerStage.plainSubSource[K, V](settings, subscription))
 
   /**
+   * The `plainPartitionedManualOffsetSource` is similar to [[#plainPartitionedSource]] but allows the use of an offset store outside
+   * of Kafka, while retaining the automatic partition assignment. When a topic-partition is assigned to a consumer, the `loadOffsetOnAssign`
+   * function will be called to retrieve the offset, followed by a seek to the correct spot in the partition. The `storeOffsetOnRevoke`
+   * function gives the consumer to chance to store
+   */
+  def plainPartitionedManualOffsetSource[K, V](settings: ConsumerSettings[K, V], subscription: AutoSubscription, loadOffsetOnAssign: TopicPartition => Long, onRevoke: TopicPartition => Unit = _ => ()): Source[(TopicPartition, Source[ConsumerRecord[K, V], NotUsed]), Control] =
+    Source.fromGraph(ConsumerStage.plainSubSource[K, V](settings, subscription, Some(loadOffsetOnAssign), onRevoke))
```

Comment:
shouldn't it be `loadOffsetOnAssign : TopicPartition => Future[Long]` so that it doesn't have to be a blocking call, e.g. when fetching from DB?

Reply:
That was my first thought too but I'm not sure it's safe. The `onPartitionAssigned` hook on the `ConsumerRebalanceListener` returns `void` and I think it expects that when it returns, the work has been done, so then I would have to do my own `Await.result` in there and I figured it would be better to have the user do it. Or maybe I'm misunderstanding how it works?

Reply:
You are already doing it async since you `consumer.tell(KafkaConsumerActor.Internal.Seek`. It wouldn't be much different to pipe the result of a Future to the ConsumerActor that will do the seek.

One corner case could be ordering, in case there are several almost simultaneous  assignments and the futures for these complete in "wrong" order.

Reply:
Ya you're right, I can probably change it to use `Future`. I'm going to have to do some more analysis to see if there is a race between when the seek happens and when the polling resumes the consumption of the topic. My suspicion is that there is one right now that I'll have to resolve.